### PR TITLE
Heatmap: Add colorScheme callback function support

### PR DESCRIPTION
**Pull request by shaswat-indian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Though `colorScheme` supports callback functions currently, but the params received there are undefined so it's unusable.

Issue Number: N/A

## What is the new behavior?
Can set color based on data/selected state of Heatmap.

https://github.com/reaviz/reaviz/assets/33908100/13bbd4f5-3e20-4430-83d6-489eb1c4f3ac

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/common/color/helper.ts [hunk 1]
```diff
@@ -118,16 +120,18 @@ const getValueScale = (
 
   return (point) => {
     // For 0 values, lets show a placeholder fill
-    if (point === undefined || point === null) {
+    if (point.value === undefined || point.value === null) {
```
Comment (ghsteff) [inline src/common/color/helper.ts:123]: a little safer
```suggestion
    if (point?.value === undefined || point?.value === null) {
```

File: src/Heatmap/HeatmapSeries/HeatmapSeries.tsx [hunk 2]
```diff
@@ -54,6 +54,11 @@ export interface HeatmapSeriesProps {
    * Cell component that will be rendered.
    */
   cell: ReactElement<HeatmapCellProps, typeof HeatmapCell>;
+
+  /**
+   * Selected cell(s) in active state
+   */
+  selectedValues?: any;
```
Comment (ghsteff) [inline src/Heatmap/HeatmapSeries/HeatmapSeries.tsx:61]: maybe we should call this `selectedPoints` to align with the `point` variable name in the helper functions

Comment (ghsteff) [reply src/Heatmap/HeatmapSeries/HeatmapSeries.tsx:61]: or maybe `activePoints` to align even _more_ with `getColor()` params heh

Comment (amcdnl) [reply src/Heatmap/HeatmapSeries/HeatmapSeries.tsx:61]: We have selection in other spots of the code - we should make the variables match that.

**Review by amcdnl — CHANGES_REQUESTED** [review]

File: src/common/color/helper.ts (hunk 1 above)
Comment (shaswat-indian) [reply src/common/color/helper.ts]: Done.

File: src/Heatmap/HeatmapSeries/HeatmapSeries.tsx (hunk 2 above)
Comment (shaswat-indian) [reply src/Heatmap/HeatmapSeries/HeatmapSeries.tsx]: Changed to `selections`.

**Review by amcdnl — APPROVED** [review]

**Pull request merged**